src/storage/txn/tablet_flow_controller.rs

```diff
@@ -0,0 +1,727 @@
+                    }
+
+                    let insert_limiter_and_checker = |region_id, suffix| -> FlowChecker<E> {
+                        let engine = tablet_factory.open_tablet_cache(region_id, suffix).unwrap();
+                        let mut v = limiters.as_ref().write().unwrap();
+                        let discard_ratio = Arc::new(AtomicU32::new(0));
+                        let limiter = v.entry(region_id).or_insert((
+                            Arc::new(
+                                <Limiter>::builder(f64::INFINITY)
+                                    .refill(Duration::from_millis(1))
+                                    .build(),
+                            ),
+                            discard_ratio,
+                        ));
+                        FlowChecker::new_with_tablet_suffix(
+                            &config,
+                            engine,
+                            limiter.1.clone(),
+                            limiter.0.clone(),
+                            suffix,
+                        )
+                    };
+                    let msg = flow_info_receiver.recv_deadline(deadline);
+                    match msg.clone() {
+                        Ok(FlowInfo::L0(_cf, _, region_id, suffix))
+                        | Ok(FlowInfo::L0Intra(_cf, _, region_id, suffix))
+                        | Ok(FlowInfo::Flush(_cf, _, region_id, suffix))
+                        | Ok(FlowInfo::Compaction(_cf, region_id, suffix)) => {
+                            let mut checkers = flow_checkers.as_ref().write().unwrap();
+                            if let Some(checker) = checkers.get_mut(&region_id) {
+                                if checker.get_tablet_suffix() != suffix {
+                                    continue;
+                                }
+                                checker.on_flow_info_msg(enabled, msg);
+                            }
+                        }
+                        Ok(FlowInfo::BeforeUnsafeDestroyRange(region_id))
+                        | Ok(FlowInfo::AfterUnsafeDestroyRange(region_id)) => {
+                            let mut checkers = flow_checkers.as_ref().write().unwrap();
+                            if let Some(checker) = checkers.get_mut(&region_id) {
+                                checker.on_flow_info_msg(enabled, msg);
+                            }
+                        }
+                        Ok(FlowInfo::Created(region_id, suffix)) => {
+                            let mut checkers = flow_checkers.as_ref().write().unwrap();
+                            let checker = checkers
+                                .entry(region_id)
+                                .or_insert_with(|| insert_limiter_and_checker(region_id, suffix));
+                            // check if the checker's engine is exactly (region_id, suffix)
+                            // if  checker.suffix < suffix, it means its tablet is old and needs the refresh
```

Comment:
Why use "suffix" to indicate the freshness?

Reply:
It's the design of tablet---basically the suffix is the raft log index. So higher suffix, the refresher the tablet.